### DISCUSSION TITLE
Unable to start connect edge on custom handle with icon/text

### DISCUSSION THREAD

**Discussion by MrRainesE** [xyflow/xyflow#2656] [Q&A] [answered]
### Issue:
custom handle with icon or text doesn't inherit onConnect start, only way start connect edge is to klik around the icon
![image](https://user-images.githubusercontent.com/76207902/206421102-fc3095d8-fc51-4e34-9f8a-cc45bca53549.png)

### expected behaviour:
I want to be able to start a connect edge on and around the icon or text.

### Codesandbox for example
https://codesandbox.io/s/async-lake-kkg07n?file=/App.js

Is there any workaround for this @moklick? based on your example here https://github.com/wbkd/react-flow/discussions/2095#discussioncomment-3396865

**Comment by bcakmakoglu** [comment, marked as answer]
You can do this to make it work on children of your Handle (in your case the icon)

```html
<Check
	className={["source"]}
	data-handleid="a"
	data-nodeid={nodeId}
	data-handlepos="right"
	style={{ width: 12, height: 12 }}
/>
```

Here's your corrected [sandbox](https://codesandbox.io/s/throbbing-snow-n57iwm?file=/ColorSelectorNode.js:1259-1445).

The issue is that the handler needs the class `source` or `target` on the child element, which is supposed to start the connection.
It also needs the handle-id, node-id and handle-pos to work properly, otherwise `isValidConnection` will not return the proper params and you might be blocked from creating new edges :)

**Reply by MrRainesE** [reply]
Thanks for your answer, but it doesn't work. i Changed the Check icon to CheckCircle so i can test this better but no result. 
I also changed the classname  from array to string, because classname only accepts strings

props get pass properly
![image](https://user-images.githubusercontent.com/76207902/207290051-1182c574-a4ff-4b31-9417-e54627b2f8ee.png)

**Reply by bcakmakoglu** [reply]
Mh.... I thought that it worked, strange. According to the handle code that should work :(
Let me check again when I got some free time, gotta do something else rn 😅

**Reply by MrRainesE** [reply]
thanks, looking forward!

**Reply by moklick** [reply]
I would add `pointerEvents: 'none'` to the icon: https://codesandbox.io/s/gallant-lehmann-vs4di0?file=/ColorSelectorNode.js:1239-1314

Is that what you are looking for?

**Reply by bcakmakoglu** [reply]
How did I miss that 🤣

**Reply by MrRainesE** [reply]
you are a genius! thanks guys
